python/lsst/ts/externalscripts/maintel/make_lsstcam_calibrations.py

```diff
@@ -198,3 +234,45 @@ def get_schema(cls):
 
     def get_instrument_configuration(self):
         return dict(filter=self.config.filter)
+
+    async def assert_feasibility(self, image_type):
+        """Ensure dome components are in the required state before flats."""
+        if image_type != "FLAT":
+            return None
+
+        # Check if dome components are ignored by user configuration.
+        if (
+            hasattr(self._mtcs.check, "mtdometrajectory")
+            and not self._mtcs.check.mtdometrajectory
+        ):
+            raise RuntimeError(
+                "Cannot take FLAT images when MTDomeTrajectory is ignored. "
+                "Dome trajectory state is required to ensure proper vignetting state publication."
+            )
+
+        if hasattr(self._mtcs.check, "mtdome") and not self._mtcs.check.mtdome:
+            raise RuntimeError(
+                "Cannot take FLAT images when MTDome is ignored. "
+                "Dome state is required to ensure proper vignetting state publication."
+            )
+
+        dome_trajectory_evt, dome_evt = await asyncio.gather(
+            self._mtcs.rem.mtdometrajectory.evt_summaryState.aget(
+                timeout=self._mtcs.long_timeout
+            ),
+            self._mtcs.rem.mtdome.evt_summaryState.aget(
+                timeout=self._mtcs.long_timeout
+            ),
+        )
+        dome_trajectory_summary_state = salobj.State(dome_trajectory_evt.summaryState)
+        dome_summary_state = salobj.State(dome_evt.summaryState)
+
+        if dome_trajectory_summary_state != salobj.State.ENABLED:
+            raise RuntimeError(
+                "MTDomeTrajectory must be ENABLED before taking FLATs to ensure "
+                "vignetting state is published."
+            )
+        if dome_summary_state not in (salobj.State.DISABLED, salobj.State.ENABLED):
```

Comment:
Instead of using a `tuple` use a `set`... Comparison on a `set` is more performant:

```
if dome_summary_state not in {salobj.State.DISABLED, salobj.State.ENABLED}:
```